websites/D/Duolingo/presence.ts

```diff
@@ -1,92 +1,232 @@
-lang.set("da", "Danish");
-lang.set("kl", "Klingon");
-lang.set("ro", "Romanian");
-lang.set("cs", "Czech");
-lang.set("sw", "Swahili");
-lang.set("cy", "Walsh");
-lang.set("id", "Indonesian");
-lang.set("hu", "Hungarian");
-lang.set("uk", "Ukrainian");
-lang.set("eo", "Esperanto");
-lang.set("nv", "Navajo");
-lang.set("en", "English");
+  language = { imageKey: "icon", name: "nothing (yet!)" };
 
-presence.on("UpdateData", async () => {
-  const path1 = document.location.pathname;
+function updateLanguage() {
+  const state = JSON.parse(window.localStorage.getItem("duo.state")),
+    courseId = state?.user?.currentCourseId;
+  if (!courseId) return;
+
+  const course = state.courses?.[courseId];
+  if (!course) return;
 
-  if (!path1.split("/")[2]) {
-    if (document.location.pathname.startsWith("/learn")) {
-      pageData.details = "Choosing level to learn..";
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.startsWith("/shop")) {
-      pageData.details = "Browsing shop..";
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/dictionary")) {
-      pageData.details = "Using dictionary..";
-      pageData.state = `Language: ${document.location.pathname.split("/")[2]}`;
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/profile")) {
-      pageData.details = "Browsing profile..";
-      pageData.state = `Browsing: ${document.location.pathname.split("/")[2]}`;
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/words")) {
-      presenceData.details = "Checking words...";
-      presenceData.largeImageKey = "logo";
-      presence.setActivity(presenceData);
-    } else if (
-      document.location.pathname === "/" ||
-      !document.location.pathname
-    )
-      presence.setActivity(pageData);
-  } else {
-    if (
-      path1.length > 1 &&
-      path1.split("/")[2] !== null &&
-      path1.split("/")[2].length === 2
-    ) {
-      let language: string;
-      for (const value of lang.keys()) {
-        if (path1.split("/")[2] === value) {
-          language = lang.get(value);
-          break;
-        }
-      }
-      presenceData.details = "Taking a lesson";
-      presenceData.state = `Language: ${language}`;
-      presenceData.largeImageKey = "logo";
-      presence.setActivity(presenceData);
+  const languageId = course.learningLanguage;
+  if (FLAG_ICONS.includes(languageId)) language.imageKey = `flag_${languageId}`;
+  else if (languageId) language.imageKey = "flag_unknown";
+  else language.imageKey = "icon";
+  language.name = course.title ?? "nothing (yet!)";
+}
+updateLanguage();
+setInterval(updateLanguage, 1000);
+
+function checkBasicPages(path: string[]) {
+  if (path.length === 0) {
+    const url = new URL(document.location.href);
+    if (url.searchParams.get("isLoggingIn") === "true") {
+      presenceData.details = "Logging in";
+      return;
```

Comment:
this for all these if statements
```suggestion
      return presenceData.details = "Logging in";
```